File: clojure/e1.clj
;Find the sum of all the multiples of 3 or 5 below 1000. (fizzbuzz++)
(defn divisible? 
[number divisor]
    (zero? (mod number divisor))
)

(defn fizzable?
[number]
    (some true? (map (partial divisible? number) '(3 5)))
)

(defn E1 
[max_num]
    (reduce + (filter fizzable? (range max_num)))
)

(println (format "%d" (E1 1000)))

File: julia/e1.jl
#= Project Euler problem 1: print the sum of all integers between 1 and 1000 (exclusive),
divisible by 3 or 5. Essentially fizzbuzz++
=#

function euler_1(max_num::Int)::Int
    reduce(+, filter(x -> x % 3 == 0 || x % 5 == 0, range(1, stop=max_num-1)))
end

println(euler_1(1000))
